src/core/stores/VNode.ts

```diff
@@ -62,94 +62,234 @@ export class VNode {
+    }
+    /**
+     * Return this VNode's next sibling.
+     *
+     */
+    get previousSibling(): VNode {
+        return this.parent && this.parent.childBefore(this);
+    }
+    /**
+     * Return this VNode's next sibling.
+     */
+    get nextSibling(): VNode {
+        return this.parent && this.parent.childAfter(this);
+    }
+
+    //--------------------------------------------------------------------------
+    // Updating
+    //--------------------------------------------------------------------------
+
+    /**
+     * Insert the given VNode before this VNode. Return self.
+     *
+     * @param node
+     */
+    before(node: VNode): VNode {
+        return this.parent.insertBefore(node, this);
+    }
+    /**
+     * Insert the given VNode after this VNode. Return self.
+     *
+     * @param node
+     */
+    after(node: VNode): VNode {
+        return this.parent.insertAfter(node, this);
     }
     /**
      * Prepend a child to this node. Return self.
      */
     prepend(child: VNode): VNode {
-        child.setPosition(this, 0);
+        return this._insertAtIndex(child, 0);
+    }
+    /**
+     * Append a child to this VNode. Return self.
+     */
+    append(child: VNode): VNode {
+        return this._insertAtIndex(child, this.children.length);
+    }
+    /**
+     * Insert the given node before the given reference (which is a child of
+     * this VNode). Return self.
+     *
+     * @param node
+     * @param reference
+     */
+    insertBefore(node: VNode, reference: VNode): VNode {
+        const index = this.indexOf(reference);
+        if (index < 0) {
+            throw new Error('The given VNode is not a child of this VNode');
+        }
+        return this._insertAtIndex(node, index);
+    }
+    /**
+     * Insert the given node after the given reference (which is a child of this
+     * VNode). Return self.
+     *
+     * @param node
+     * @param reference
+     */
+    insertAfter(node: VNode, reference: VNode): VNode {
+        const index = this.indexOf(reference);
+        if (index < 0) {
+            throw new Error('The given VNode is not a child of this VNode');
+        }
+        return this._insertAtIndex(node, index + 1);
+    }
+    /**
+     * Remove the given child from this VNode. Return self.
+     *
+     * @param child
+     */
+    removeChild(child: VNode): VNode {
+        const index = this.indexOf(child);
+        if (index < 0) {
+            throw new Error('The given VNode is not a child of this VNode');
+        }
+        return this._removeAtIndex(index);
+    }
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Insert a VNode at the given index within this VNode's children.
+     * Return self.
+     *
+     * @param child
+     * @param index
+     */
+    _insertAtIndex(child: VNode, index: number): VNode {
```

Comment:
This should start with:
```typescript
if (child.parent) {
    child.parent.removeChild(child);
}
```
This way if we insert an element that already is in the `vDocument`, it's moved, not duplicated.